chef/automate
# Embed latest nats-server in event-gw and event-svc

### :nut_and_bolt: Description: What code changed, and why?

We had some patches to nats-server to support mutual TLS, which have
since been merged. In the meantime, the nats project shipped version 2
(which was very soon after our initial import/vendor) and we moved
automate from the `dep` tool to modules. This patch contains the
necessary changes to use the 2.x nats-server code in the event-gateway
and event-service. End-to-end functionality was verified using the
docker-based hab-client from applications-service's dev/ dir.

### :chains: Related Resources

#2256

### :+1: Definition of Done

No longer pulling nats server from Dan's branch. ✅ 

### :athletic_shoe: How to Build and Test the Change

* Build event-service and event-gateway
* start all services
* the development data generator still works: run `applications_populate_database` and see the data in the UI
* Works e2e with a real hab client: follow the instructions here: https://github.com/chef/automate/blob/master/components/applications-service/README.md
